Subject: make 'core show settings' should show all settable directories
From: Tzafrir Cohen <tzafrir.cohen@xorcom.com>
Origin: http://svn.digium.com/view/asterisk?view=rev&revision=254162

As we give a different value to datadir, we better make such settings
viewable at run-time. This patch makes all those settings available through
the Asterisk CLI command 'core show settings'.

Included in trunk (to be 1.8). Not included in upstream 1.6.2 but applies
cleanly.

diff --git a/main/asterisk.c b/main/asterisk.c
index 3b0c7c9..93d52c0 100644
--- a/main/asterisk.c
+++ b/main/asterisk.c
@@ -482,6 +482,13 @@ static char *handle_show_settings(struct ast_cli_entry *e, int cmd, struct ast_c
 	ast_cli(a->fd, "  Module directory:            %s\n", ast_config_AST_MODULE_DIR);
 	ast_cli(a->fd, "  Spool directory:             %s\n", ast_config_AST_SPOOL_DIR);
 	ast_cli(a->fd, "  Log directory:               %s\n", ast_config_AST_LOG_DIR);
+	ast_cli(a->fd, "  Run/Sockets directory:       %s\n", ast_config_AST_RUN_DIR);
+	ast_cli(a->fd, "  PID file:                    %s\n", ast_config_AST_PID);
+	ast_cli(a->fd, "  VarLib directory:            %s\n", ast_config_AST_VAR_DIR);
+	ast_cli(a->fd, "  Data directory:              %s\n", ast_config_AST_DATA_DIR);
+	ast_cli(a->fd, "  ASTDB:                       %s\n", ast_config_AST_DB);
+	ast_cli(a->fd, "  IAX2 Keys directory:         %s\n", ast_config_AST_KEY_DIR);
+	ast_cli(a->fd, "  AGI Scripts directory:       %s\n", ast_config_AST_AGI_DIR);
 	ast_cli(a->fd, "\n\n");
 	return CLI_SUCCESS;
 }
